modules/400-descheduler/hooks/migrate_from_cm.go

```diff
@@ -0,0 +1,150 @@
+				},
+				TargetThresholds: map[v1.ResourceName]v1alpha1.Percentage{
+					"cpu":    80,
+					"memory": 90,
+					"pods":   90,
+				},
+			},
+		}
+	}
+	if value := config.Get("highNodeUtilization"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.HighNodeUtilization = &v1alpha1.HighNodeUtilization{
+			NodeResourceUtilizationThresholds: &v1alpha1.NodeResourceUtilizationThresholdsFiltering{
+				Thresholds: map[v1.ResourceName]v1alpha1.Percentage{
+					"cpu":    50,
+					"memory": 50,
+				},
+			},
+		}
+	}
+	if value := config.Get("removePodsViolatingInterPodAntiAffinity"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.RemovePodsViolatingInterPodAntiAffinity = &v1alpha1.RemovePodsViolatingInterPodAntiAffinity{}
+	}
+	if value := config.Get("removePodsViolatingNodeAffinity"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.RemovePodsViolatingNodeAffinity = &v1alpha1.RemovePodsViolatingNodeAffinity{
+			NodeAffinityType: []string{"requiredDuringSchedulingIgnoredDuringExecution"},
+		}
+	}
+	if value := config.Get("removePodsViolatingNodeTaints"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.RemovePodsViolatingNodeTaints = &v1alpha1.RemovePodsViolatingNodeTaints{}
+	}
+	if value := config.Get("removePodsViolatingTopologySpreadConstraint"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.RemovePodsViolatingTopologySpreadConstraint = &v1alpha1.RemovePodsViolatingTopologySpreadConstraint{}
+	}
+	if value := config.Get("removePodsHavingTooManyRestarts"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.RemovePodsHavingTooManyRestarts = &v1alpha1.RemovePodsHavingTooManyRestarts{
+			PodsHavingTooManyRestarts: &v1alpha1.PodsHavingTooManyRestartsParameters{
+				PodRestartThreshold:     100,
+				IncludingInitContainers: true,
+			},
+		}
+	}
+	if value := config.Get("podLifeTime"); value.Exists() {
+		deschedulerCR.Spec.DeschedulerPolicy.Strategies.PodLifeTime = &v1alpha1.PodLifeTime{
+			PodLifeTime: &v1alpha1.PodLifeTimeParameters{
+				MaxPodLifeTimeSeconds: uintPtr(86400),
+				PodStatusPhases:       []string{"Pending"},
+			},
+		}
+	}
+
+	if value := config.Get("nodeSelector"); value.Exists() {
+		rawSelectorMap := value.Map()
+		nodeSelectorSet := make(labels.Set, len(rawSelectorMap))
+		for k, v := range rawSelectorMap {
+			nodeSelectorSet[k] = v.String()
+		}
+
+		deschedulerCR.Spec.DeploymentTemplate.NodeSelector = nodeSelectorSet
+	}
+
+	if value := config.Get("tolerations"); value.Exists() {
+		var tolerations []v1.Toleration
+
+		err := mapstructure.Decode(value.Value(), &tolerations)
+		if err != nil {
+			return fmt.Errorf("can't decode existing tolerations %+v: %s", value.Value(), err)
+		}
+
+		deschedulerCR.Spec.DeploymentTemplate.Tolerations = tolerations
+	}
+
+	input.PatchCollector.Create(deschedulerCR, object_patch.UpdateIfExists())
+
+	input.ConfigValues.Set("descheduler", map[string]string{})
```

Comment:
Maybe it is better to use `nil` here?

Reply:
The target state is that there is no descheduler key in the deckhouse configmap.